### DISCUSSION TITLE
OpenBSD

### DISCUSSION THREAD

**Discussion by MSaLeM18** [vlang/v#17192] [Installation Issues]
**In OpenBSD installation of GC ([`BDWGC`](https://github.com/ivmai/bdwgc)) must be installed as a prerequisite since it's not installed by default ...**

### OpenBSD, within V repo:
```
doas pkg_add boehm-gc-8.0.4p0 
CC=cc gmake
```

**Comment by j-bm** [comment]
You can omit the version, as pkg_add will fill that in automatically.  (In future releases of OpenBSD and boehm-gc the version may change.)

**Comment by JalonSolov** [comment]
V includes boehm-gc in the `thirdparty/libgc` directory, and will compile it for your system.  You only need to install it from the package manager if you can't/don't want to use the one V supplies.

**Reply by vetelko** [reply]
getting following error, how can I achieve V compiles libgc from thirdpart/libgc for me?
....
V has been successfully built
./v run ./cmd/tools/detect_tcc.v
V panic: >> error: none of the paths ['/usr/local/lib/libgc.a', '/usr/lib/libgc.a'] exist
....

**Reply by j-bm** [reply]
For OpenBSD you need the OpenBSD-supplied Boehm GC package.  As root:

`# pkg_add boehm-gc`

The hint as to why this is required is in your V panic message: it looks for `/usr/local/lib/something` which is the usual path for add-on packages in OpenBSD.

**Reply by vetelko** [reply]
Thank you j-bm, I only reacted to @JalonSolov comment that v can compile its own version so adding boehm-gc package is not needed.

**Comment by j-bm** [comment]
Actually the OpenBSD package for boehm-gc has specific patches to make it work on that platform.  (OpenBSD 7.2 has boehm-gc version 8.0.4 plus patches).

I haven't compared the two versions (the thirdparty/ version appears to be from a sqlite3 build system.)  

For reference, the OpenBSD port of boehm-gc includes support for mmap, threads using pthreads, C++, static libraries and support for hardware-atomic ops in a separate library.

In my own experience, adding the boehm-gc package was required for using V on OpenBSD as somehow the thirdparty/ version was not used.
